### ISSUE TITLE
Warning! Error save data

### ISSUE THREAD

**Issue by rumytaulu** [ridwanskaterock/cicool#205]
I got a "warning! error save data" when I want to save the new crud I've just created.
Any suggestion to fix this?

[Indonesian]
Saya mendapat pesan "warning! error save data" saat hendak menyimpan new crud yang baru dibuat
Ada saran untuk memperbaiki?
![screenshot from 2018-07-18 16-25-19](https://user-images.githubusercontent.com/14027018/42869490-c58229cc-8aa7-11e8-87f3-bc0f9ed912f1.png)

**Comment by ridwanskaterock** [comment]
can you check response for XHR to see detail error 
![image](https://user-images.githubusercontent.com/9270445/42936339-017a5b88-8b76-11e8-95d2-a2024aecd476.png)

Thanks

**Comment by rumytaulu** [comment]
Here's the Response :

html | <div class="table-wrapper">  <table class="table table-responsive table table-bordered  table-striped"  id="diagnosis_list">    <thead>       <tr>                            <th width="20" rowspan="2" valign="midle"  style="vertical-align: middle; text-align: center;">No</th>                            <th width="120" rowspan="2" valign="midle"  style="vertical-align: middle; text-align:  center;">Field</th>                            <th width="80" colspan="4"  style="text-align: center;">Show On</th> …alue="in_list"  class="input  text  number  select  options"  data-group-validation="input  text  number  select  options"  data-placeholder="" title="yes">In List</option>                                                <option  value="valid_ip" class="input  text" data-group-validation="input  text"  data-placeholder="" title="no">Valid Ip</option>                                            </select>                </div>             </div>          </td>       </tr>          </tbody> </table> </div>
-- | --
subject | Data Register
success | true
![screenshot from 2018-07-20 00-57-39](https://user-images.githubusercontent.com/14027018/42958369-f0a1b6e8-8bb7-11e8-98c7-fb1f0f39f060.png)

**Comment by ridwanskaterock** [comment]
please share XHR response only error 
![image](https://user-images.githubusercontent.com/9270445/42980531-f8a2bdac-8c01-11e8-941d-acf1d7660a5e.png)

Thanks

**Comment by rumytaulu** [comment]
![screenshot from 2018-07-20 11-48-59](https://user-images.githubusercontent.com/14027018/42982223-ef2bc26c-8c12-11e8-90d8-1deeac21c558.png)

**Comment by ridwanskaterock** [comment]
can you copy raw response  ? 

what you r PHP version ? 

thanks

**Comment by rumytaulu** [comment]
FYI ... this is the first time I've experience this kind of error, every cicool app I generated was working perfectly.

This is my PHP version : 
PHP 7.0.30-0ubuntu0.16.04.1

RAW Response : 

Cache-Control	
no-store, no-cache, must-revalidate
Connection	
close
Content-Type	
application/json
Date	
Fri, 20 Jul 2018 03:47:10 GMT
Expires	
Thu, 19 Nov 1981 08:52:00 GMT
Pragma	
no-cache
Server	
Apache/2.4.18 (Ubuntu)
Set-Cookie	
__osk8cockkkoc48wcks8wc8ggco8k…x-Age=86400; path=/; HttpOnly
Transfer-Encoding	
chunked
Request headers (618 B)	
Accept	
application/json, text/javascript, */*; q=0.01
Accept-Encoding	
gzip, deflate
Accept-Language	
en-GB,en;q=0.5
Connection	
keep-alive
Content-Length	
20839
Content-Type	
application/x-www-form-urlencoded; charset=UTF-8
Cookie	
language=english; language_sys…g76d2t66ie91o0j1c7gd5f5ej0ph4
Host	
register
Referer	
http://register/administrator/crud/add
User-Agent	
Mozilla/5.0 (X11; Ubuntu; Linu…) Gecko/20100101 Firefox/61.0
X-Requested-With	
XMLHttpRequest

![screenshot from 2018-07-20 13-11-54](https://user-images.githubusercontent.com/14027018/42984400-f1492c5e-8c1e-11e8-83af-3e74f59f382e.png)

**Comment by rumytaulu** [comment]
[English]
Sorry for the late comment, kinda busy.
I've solved this.
The problem is some of the table's column didn't have Collation ("utf8_general_ci in my case), I converted the table in question from excel worksheet, maybe that's why it ended with BLANK COLLATION, which prevent Cicool for creating the new CRUD and generating the error.

[Indonesian]
Maaf baru membalas komentar, agak sibuk akhir-akhir ini.
Masalahnya sudah berhasil dipecahkan.
Beberapa kolom di tabel rupanya tidak mempunya Collation. Tabelnya di konversi dari Excel Worksheet, mungkin itu sebabnya hasilnya beberapa kolom tabel tidak punya Collation (Collationnya KOSONG), yang berakibat CRUD tidak bisa dibuat dan menghasilkan error.
